src/main/java/webserver/HttpRequest.java

```diff
@@ -1,47 +1,51 @@
 package webserver;
 
-import util.HttpRequestUtils;
-import util.HttpRequestUtils.Pair;
-
+import java.io.BufferedReader;
 import java.io.IOException;
 import java.net.URLDecoder;
 import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
+import util.HttpRequestUtils;
+import util.IOUtils;
 
 public class HttpRequest {
 
-    private String httpMethod;
-    private String httpUrl;
-    private String httpVersion;
-    private Map<String, String> params;
-    private List<Pair> headers;
+    private BufferedReader br;
+    private RequestLine requestLine;
+    private Map<String, String> headers;
+    private Map<String, String> parameters;
 
-    public HttpRequest(String requestLine, List<Pair> headers) throws IOException {
+    public HttpRequest(BufferedReader br) throws IOException {
+        this.br = br;
+        init();
+    }
+
+    private void init() throws IOException {
+        String requestLine = URLDecoder.decode(br.readLine(), StandardCharsets.UTF_8);
         String[] requestLineSplit = requestLine.split(" ");
-        this.httpMethod = requestLineSplit[0];
-        this.httpUrl = requestLineSplit[1];
-        this.httpVersion = requestLineSplit[2];
-        this.params = parseParams(httpUrl);
-        this.headers = headers;
+        this.requestLine = new RequestLine(requestLineSplit[0], requestLineSplit[1], requestLineSplit[2]);
```

Comment:
String을 받아서 쪼개는 일을 requestLine 객체에서 하면 좋을 것 같습니다.
```suggestion
        this.requestLine = new RequestLine(requestLine);
```